doc/user/content/ingest-data/segment.md

```diff
@@ -0,0 +1,139 @@
+
+In the code, you will notice a `CHECK` statement. It defines how to validate each request. At the time of writing, Segment
+validates requests by signing them with an HMAC in the `X-Signature` request header. The HMAC is a
+hex-encoded SHA1 hash using the shared secret from the previous step and request body. Materialize decodes the signature using
+the [`decode`](/sql/functions/#decode) function, getting the raw bytes, and generate our own HMAC
+using the [`hmac`](/sql/functions/#hmac) function. If the two values are equal, then the request is
+legitimate!
+
+## Step 4. Create a Webhook Destination in Segment
+
+1. In Segment, go to **Connections > Catalog**.
+
+2. Search and click **Webhooks (Actions)**.
+
+3. Click **Add destination**.
+
+4. Select a data source and click **Next**.
+
+5. Jot a *Destination Name* and click **Create Destination**.
+
+## Step 5. Configure the Mappings in Segment
+
+A webhook destination in Segment requires a [data mapping](https://segment.com/blog/data-mapping/) to send events from the source to the destination. For this guide, the destination is the Materialize source. Follow these steps to create the correct mapping:
+
+1. Go to your webhook destination created in the previous step.
+
+2. Click on **Mappings > New Mapping**.
+
+3. Click **Send** and fill the configuration as follows:
+   1. In **Select events to map and send**, fill the conditions as you wish.
+   2. In **Add test**, select any of the events.
+   3. In **Select mappings**, fill the fields as follows:
+    Field               | Value
+    ------------------- | ----------------------
+    Url                 | Use the URL from the [Step 3](`/`).
+    Method              | `POST`
+    Batch Size          | `0`
+    Headers             | -
+    Data                | `$event`
+    Enable Batching     | `No`.
+4. Click **Test Mapping** and validate the webhook is working.
+5. After a succesful test, click **Save**.
+
+  {{< note >}}
+  If **Test Mapping** fails in and throws a *"failed to validate the request"* error, it means te shared secret is not right. To fix the error, follow this steps:
+ 1. In **Segment**, go to your webhook destination created in the **Step 4**.
+ 2. Click **Settings**.
+ 3. In **Shared Secret**, enter the secret created in the **Step 2**.
+ 4. Click **Save Changes**.
+  {{< /note >}}
+
+## Step 5. Parse Incoming Data
+
+Create a materialized view to parse the incoming events from Segment:
+
+```sql
+CREATE VIEW segment_source_parsed AS
+WITH parse AS (
+    SELECT
+        body->>'email' AS email,
+        body->>'event' AS event,
+        body->>'messageId' AS message_id,
+        body->>'properties' AS properties,
+        body->>'timestamp' AS ts,
+        body->>'type' AS event_type,
+        body->>'userId' AS user_id
+    FROM my_segment_source
+);
+```
```

Comment:
It looks like you should drop the with clause.

Reply:
Thank you for pointing that out! @bobbyiliev we should apply the same in RudderStack